src/python_routing_v02/fast_reach/mc_reach.pyx

```diff
@@ -389,9 +391,14 @@ cpdef object compute_network(
                 # Update indexes to point to next reach
                 ireach_cache += reachlen
                 iusreach_cache += usreachlen
-
-            timestep += 1
-
+            with gil:
+                if nudging_positions_list != []:
```

Comment:
This should be:
```python
if nudging_positions_list:
    ...
```
Which means "if nudging_positions_list is truthful (ie not empty)"